nodestream/project/pipeline_scope.py

```diff
@@ -26,26 +26,34 @@ def __init__(
         pipelines: List[PipelineDefinition],
         persist: bool = True,
         config: ScopeConfig = None,
-        targets: List[str] = None,
+        targets: List[str] = [],
     ) -> None:
         self.persist = persist
         self.name = name
         self.config = config
         self.targets = targets
         self.pipelines_by_name: Dict[str, PipelineDefinition] = {}
         for pipeline in pipelines:
+            if pipeline.targets is not None:
+                pipeline.targets = list(set(pipeline.targets) | set(self.targets))
```

Comment:
This makes we think that maybe we should store the `target` as a `set` to begin with. What do you think?